### PR TITLE
Fixed issue-  Customer non-default addresses' post codes do not change on leading zero

### PR DESCRIPTION
**Pull request by shikhamis11**
Fixed issue #20100 and #17452

### Preconditions

1. Magento 2.2.4
2. Create a customer with at least 3 addresses.  The non-default address should have a post code that is only 4 numbers.

### Steps to reproduce

1. Edit the customer addresses in the admin panel, and add a leading zero to the post code of the address that is not a default billing or shipping address.
2. Save the customer.
3. Look at the address to confirm the post code did not change.

### Expected result

1. Post code should change to whatever was entered.

### Actual result

1. Post code does not change.

I tested and this works the opposite way as well:  If the post code has a leading zero and you remove it and save, it retains the leading zero.  So, it seems to be equating 01111 = 1111 and ignoring it, regardless of which side of the equation it's on.  You can also change it to an entirely different value (e.g. 22222) and that works as expected.  The issue is just changing it from a leading zero or to a leading zero, and only on non-default addresses.

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by davidverholen — APPROVED** [review]

**Comment by orlangur** [comment]
Hi @shikhamis11, could you please explain what did you try to achieve by requesting review from me and Vladimir?

**Comment by shikhamis11** [comment]
@orlangur that was not intentionally but I had doubt whether it will process further towards merging or not

**Comment by shikhamis11** [comment]
@orlangur can you tell me when Pull requests goes to Todo only and when ticket created and PRs goes to Todo as well as testing in progress in community pull request after approval from maintainers

**Comment by sivaschenko** [comment]
Performance tests results:

    Simple Product 2 Add To Cart (Add To Cart By Guest): 75% (mainline = 4 | team = 7) -> DEGRADATION
    Configurable Product 1 Add To Cart (Add To Cart By Guest): 125% (mainline = 4 | team = 9) -> DEGRADATION
    Simple Product 2 Add To Cart (Checkout By Guest): 75% (mainline = 4 | team = 7) -> DEGRADATION
    Configurable Product 1 Add To Cart (Checkout By Guest): 125% (mainline = 4 | team = 9) -> DEGRADATION
    Checkout Billing/Shipping Information (Checkout By Guest): 325% (mainline = 4 | team = 17) -> DEGRADATION
    Checkout Payment Info/Place Order (Checkout By Guest): 175% (mainline = 8 | team = 22) -> DEGRADATION
    Simple Product 1 Add To Cart (Checkout By Customer): 25% (mainline = 4 | team = 5) -> DEGRADATION
    Simple Product 2 Add To Cart (Checkout By Customer): 75% (mainline = 4 | team = 7) -> DEGRADATION
    Configurable Product 1 Add To Cart (Checkout By Customer): 125% (mainline = 4 | team = 9) -> DEGRADATION
    Checkout Billing/Shipping Information (Checkout By Customer): 325% (mainline = 4 | team = 17) -> DEGRADATION
    Checkout Payment Info/Place Order (Checkout By Customer): 162.5% (mainline = 8 | team = 21) -> DEGRADATION
    Simple Product 1 Add To Cart (Add To Cart By Customer): 25% (mainline = 4 | team = 5) -> DEGRADATION
    Simple Product 2 Add To Cart (Add To Cart By Customer): 75% (mainline = 4 | team = 7) -> DEGRADATION
    Configurable Product 1 Add To Cart (Add To Cart By Customer): 125% (mainline = 4 | team = 9) -> DEGRADATION
    Remove item (Add To Cart By Customer): 300% (mainline = 4 | team = 16) -> DEGRADATION
    Remove item (Add To Cart By Customer) - Another one - Another one: 250% (mainline = 4 | team = 14) -> DEGRADATION
    Remove item (Add To Cart By Customer) - Another one - Another one - Another one - Another one: 200% (mainline = 3 | team = 9) -> DEGRADATION
    Add Products (Admin Create Order): 33.3% (mainline = 3 | team = 4) -> DEGRADATION
    Collect Shipping Rates (Admin Create Order): 30.8% (mainline = 13 | team = 17) -> DEGRADATION
    Save Order (Admin Create Order): 133.3% (mainline = 12 | team = 28) -> DEGRADATION
    Add product to quote (API Checkout): 50% (mainline = 2 | team = 3) -> DEGRADATION
    Add product to Guest Cart (API Checkout): 50% (mainline = 2 | team = 3) -> DEGRADATION
    Add Gift Message to Guest Cart (API Checkout): 100% (mainline = 0 | team = 2) -> DEGRADATION
    Checkout Billing/Shipping Information (API Checkout): 100% (mainline = 3 | team = 6) -> DEGRADATION
    Checkout Payment Info/Place Order (API Checkout): 42.9% (mainline = 7 | team = 10) -> DEGRADATION
    Set Billing Address On Cart (GraphQL Set Billing Address On Cart): 50% (mainline = 2 | team = 3) -> DEGRADATION
    Add Simple Product To Cart (GraphQL Add Simple Product To Cart): 50% (mainline = 2 | team = 3) -> DEGRADATION
    Add Configurable Product To Cart (GraphQL Add Configurable Product To Cart): 50% (mainline = 2 | team = 3) -> DEGRADATION
    Add Simple Product To Cart (GraphQL Update Simple Product Qty In Cart): 50% (mainline = 2 | team = 3) -> DEGRADATION
    Update Simple Product qty In Cart (GraphQL Update Simple Product Qty In Cart): 20% (mainline = 5 | team = 6) -> DEGRADATION
    Add Configurable Product To Cart (GraphQL Update Configurable Product Qty In Cart): 50% (mainline = 2 | team = 3) -> DEGRADATION
    Update Configurable Product qty In Cart (GraphQL Update Configurable Product Qty In Cart): 20% (mainline = 5 | team = 6) -> DEGRADATION
    Add Simple Product To Cart (GraphQL Remove Simple Product From Cart): 50% (mainline = 2 | team = 3) -> DEGRADATION
    Remove Simple Product From Cart (GraphQL Remove Simple Product From Cart): 66.7% (mainline = 3 | team = 5) -> DEGRADATION
    Add Configurable Product To Cart (GraphQL Remove Configurable Product From Cart): 50% (mainline = 2 | team = 3) -> DEGRADATION
    Remove Configurable Product From Cart (GraphQL Remove Configurable Product From Cart): 66.7% (mainline = 3 | team = 5) -> DEGRADATION
    Add Simple Product To Cart (GraphQL Apply Coupon To Cart): 50% (mainline = 2 | team = 3) -> DEGRADATION
    Apply Coupon To Cart (GraphQL Apply Coupon To Cart): 25% (mainline = 4 | team = 5) -> DEGRADATION
    Add Simple Product To Cart (GraphQL Remove Coupon From Cart): 50% (mainline = 2 | team = 3) -> DEGRADATION
    Apply Coupon To Cart (GraphQL Remove Coupon From Cart): 25% (mainline = 4 | team = 5) -> DEGRADATION
    Remove Coupon From Cart (GraphQL Remove Coupon From Cart): 25% (mainline = 4 | team = 5) -> DEGRADATION
    Add Configurable Product To Cart (GraphQL Checkout By Guest): 50% (mainline = 2 | team = 3) -> DEGRADATION
    Add Simple Product To Cart (GraphQL Checkout By Guest): 100% (mainline = 3 | team = 6) -> DEGRADATION
    Set Billing Address On Cart (GraphQL Checkout By Guest): 250% (mainline = 2 | team = 7) -> DEGRADATION
    Set Payment Method On Cart (GraphQL Checkout By Guest): 600% (mainline = 1 | team = 7) -> DEGRADATION
    Set Shipping Method On Cart (GraphQL Checkout By Guest): 166.7% (mainline = 3 | team = 8) -> DEGRADATION
    Apply Coupon To Cart (GraphQL Checkout By Guest): 16.7% (mainline = 6 | team = 7) -> DEGRADATION
    Remove Coupon From Cart (GraphQL Checkout By Guest): 16.7% (mainline = 6 | team = 7) -> DEGRADATION

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @shikhamis11 please see the performance tests results in the comment above. I'd suggest probably trying to move the fix closer to business logic.

**Comment by sidolov** [comment]
@shikhamis11 , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**
